src/db/sysdb_subdomains.c

```diff
@@ -377,11 +377,20 @@ errno_t sysdb_update_subdomains(struct sss_domain_info *domain,
         id = ldb_msg_find_attr_as_string(res->msgs[i],
                                          SYSDB_SUBDOMAIN_ID, NULL);
 
-        is_mpg = ldb_msg_find_attr_as_bool(res->msgs[i],
-                                           SYSDB_SUBDOMAIN_MPG, false);
-        if (is_mpg) {
+        str_mpg_mode = ldb_msg_find_attr_as_string(res->msgs[i],
+                                                   SYSDB_SUBDOMAIN_MPG, NULL);
+        if (str_mpg_mode == NULL || *str_mpg_mode == '\0') {
+            str_mpg_mode = "false";
+        }
+
+        if (strcasecmp(str_mpg_mode, "FALSE") == 0) {
+            mpg_mode = MPG_DISABLED;
+        } else if (strcasecmp(str_mpg_mode, "TRUE") == 0) {
             mpg_mode = MPG_ENABLED;
         } else {
+            DEBUG(SSSDBG_MINOR_FAILURE,
+                  "Invalid value for %s\n, assuming disabled",
+                  SYSDB_SUBDOMAIN_MPG);
```

Comment:
There is a bit of inconsistency between reading the mpg_mode here and earlier in src/tests/sysdb-tests.c:952. Here we assume no MPG on error, earlier we fail with unknown MPG value. So far this is in line with how the code behaved earlier, but also if the reviewer feels like this should be harmonized, I'm OK with that.